pkg/cache/cache_test.go

```diff
@@ -297,3 +297,61 @@ func (s *testRegionCacheSuite) TestTwoQueueCache(c *C) {
 	c.Assert(ok, IsFalse)
 	c.Assert(val, IsNil)
 }
+
+var _ PriorityQueueItem = PriorityQueueItemTest(0)
+
+type PriorityQueueItemTest uint64
+
+func (pq PriorityQueueItemTest) ID() uint64 {
+	return uint64(pq)
+}
+
+func (s *testRegionCacheSuite) TestPriorityQueue(c *C) {
+	testData := []PriorityQueueItemTest{0, 1, 2, 3, 4, 5}
+	pq := NewPriorityQueue(0)
+	c.Assert(pq.Put(1, testData[1]), IsFalse)
+
+	// it will have priority-value pair as 1-1 2-2 3-3
+	pq = NewPriorityQueue(3)
+	c.Assert(pq.Put(1, testData[1]), IsTrue)
+	c.Assert(pq.Put(2, testData[2]), IsTrue)
+	c.Assert(pq.Put(3, testData[4]), IsTrue)
+	c.Assert(pq.Put(5, testData[4]), IsTrue)
+	c.Assert(pq.Put(6, testData[5]), IsFalse)
+	c.Assert(pq.Put(3, testData[3]), IsTrue)
```

Comment:
Does the same priority lead to frequent swaps in and out? is it as expect?

Reply:
it will swap only if the lowest priority less than the input's priority. As this case, after the 3->4 will become  5->4, it will not be swapped by 6->5 because the 6>5, but it will be swapped by the 3->3 because 3<5.

Reply:
I mean same priority, actually, if you use miss peer count as the priority, there are many regions have the same priority.

Reply:
yes, I will change the input condition(!min.Less(entry)) and add unit test case